variables.tf

```diff
@@ -216,12 +216,27 @@ variable "kms_key_crn" {
   type        = string
   description = "(Optional) The root key CRN of a Key Management Service like Key Protect or Hyper Protect Crypto Service (HPCS) that you want to use for disk encryption. If `null`, database is encrypted by using randomly generated keys. See https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-key-protect&interface=ui#key-byok for current list of supported regions for BYOK"
   default     = null
+  validation {
+    condition = anytrue([
+      var.kms_key_crn == null,
+      can(regex(".*kms.*", var.kms_key_crn)),
+      can(regex(".*hs-crypto.*", var.kms_key_crn))
+    ])
+    error_message = "A Key Protect or Hyper Protect key is required for encryption"
```

Comment:
This error message is misleading. As per the description `kms_key_crn ` can be `null` as well and in that case, database encryption is done using the random key. Hence Key Protect or Hyper Protect keys are not mandatory to have.

Reply:
I have opted for

    error_message = "(Optional) The root key CRN of a Key Management Service like Key Protect or Hyper Protect Crypto Service (HPCS) that you want to use for disk encryption."
    
The error does not include the part about what happens for null, because if the value is null the error will not be displayed. Is this good, or could further changes make it better?

Reply:
I think we don't need this "validation" block at all. `main.tf` validation should suffice.

Reply:
The validation here just looks at potential values for these properties, key crn must be null, kms, or hs-crypto; backup key crn must be null or kms. The main.tf validates multiple properties together, the key crn must not be null when authorization policy is created.